Newsletters: add 268 (2023-09-13)

harding opened the pull request:
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin Core PR Review Club @LarryRuane 
- [x] Bitcoin Core 26567 @murchandamus

bitschmidty commented on _topics/en/client-side-validation.md:
```diff
@@ -0,0 +1,98 @@
+
+**[Taproot Assets][]**, formerly called **Taro**, is a protocol heavily
+inspiried by RGB that uses [taproot][topic taproot]'s commitment
+structure to make allow transactions to commit to additional data.
```
```suggestion
structure to allow transactions to commit to additional data.
```

bitschmidty commented on _topics/en/client-side-validation.md:
```diff
@@ -0,0 +1,98 @@
+inspiried by RGB that uses [taproot][topic taproot]'s commitment
+structure to make allow transactions to commit to additional data.
+Taproot's construction itself derives from pay-to-contract.  As the name
+suggests, initial protocol developemnt is specifically focused on the
```
```suggestion
suggests, initial protocol development is specifically focused on the
```

bitschmidty commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+answers.  Click on a question below to see a summary of the answer from
+the meeting.*
+
+[transport abstraction][review club 28165] is a PR by Pieter Wuille (sipa)
```
```suggestion
[Transport abstraction][review club 28165] is a PR by Pieter Wuille (sipa)
```

bitschmidty commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+the meeting.*
+
+[transport abstraction][review club 28165] is a PR by Pieter Wuille (sipa)
+that introduces a _transport_ absraction (interface class). Concrete
```
```suggestion
that introduces a _transport_ abstraction (interface class). Concrete
```

bitschmidty commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+deserialization. These classes do not do the actual sending and receiving.
+
+The PR derives two concrete classes from the `Transport` class,
+`V1transport` (what we have today) and `V2Transport` (encrypted on the
```
```suggestion
`V1Transport` (what we have today) and `V2Transport` (encrypted on the
```

bitschmidty commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+`V1transport` (what we have today) and `V2Transport` (encrypted on the
+wire). This PR, which was recently merged, is part of the
+[BIP324][bip-transport-v2]
+_Version 2 P2P Encrypted Transport Protocol_ project.
```
Suggest a topic link to https://bitcoinops.org/en/topics/v2-p2p-transport/

bitschmidty replied:
Might also link part of the text ("project"?) to the tracking issue: https://github.com/bitcoin/bitcoin/issues/27634

bitschmidty commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+
+The PR derives two concrete classes from the `Transport` class,
+`V1transport` (what we have today) and `V2Transport` (encrypted on the
+wire). This PR, which was recently merged, is part of the
```
It might be nice to work this bip324 context into the introductory sentence to help readers calibrate

bitschmidty commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+  q5="In the process of turning an application object like a
+      `CTransactionRef` (transaction) into bytes / network packets, what
+      happens? Which data structures does it turn into in the process?"
+  a4="`msgMaker.Make()` serializes the `CTransactionRef` message by
```
```suggestion
  a5="`msgMaker.Make()` serializes the `CTransactionRef` message by
```

bitschmidty commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+  q7="After `PushMessage()` returns, have we sent the bytes corresponding
+      to this message to the peer already (yes/no/maybe)? Why?"
+  a7="All are possible. **Yes**: we (*net_processing*) don't have to do
+      anything else to get it to go. **No**: it's extremely unlikely to have
```
"to get it to go" could be perhaps clarified? (I dont get what it means personally)

harding commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,186 @@
+      deserialization that causes the node to reject a consensus-valid
+      block, also a soft fork."
+  a3link="https://bitcoincore.reviews/28165#l-45"
+
```
Although instagibbs calls the <4MB theoretical bug a soft fork, neither glozow nor anyone else I see calls the generalized deserialization problem a soft fork.  I'm not sure everything in that class of failures would be a soft fork; e.g., a non-deterministic deserialization failure (kinda similar to the BIP50 failure) wouldn't be a soft fork; more relevant, the type of platform-inconsistent serialization/deserialization behavior that BIP66 surreptitiously fixed isn't something I'd have described as a soft fork if it had been triggered and caused a consensus failure.

In other words, I think we should drop the phrase ", also a soft fork" from the end of this paragraph.  (But I don't have strong feelings about it.)

LarryRuane replied:
So, to be designated a soft fork, the behavior change in question must be intentional?

But I'm fine with dropping both mentions of "soft fork" if it's at all confusing, I think the answer will still be understandable and accurate.

harding replied:
@LarryRuane 

> So, to be designated a soft fork, the behavior change in question must be intentional?

Maybe?  But the point I'm trying to make is that, to be designated a soft fork, the behavior needs to be consistent.  If the behavior between nodes is inconsistent, it's a _consensus failure_.  We might eventually resolve a consensus failure using a soft fork (e.g., BIP50 resolved the BDB locks inconsistency with a soft fork that temporarily reduced the maximum block size below the level where the maximum lock count was reached, and BIP66 resolved the DER decoding inconsistency with a stricter decoding rule) but my point is that a bug in the (de)serialization code wouldn't necessarily produce a soft fork because it wouldn't necessarily occur consistently between full nodes, so it could cause a consensus failure (and consensus failures are not soft forks, at least in my personal terms sheet).

LarryRuane replied:
That's an excellent way to look at it, thanks, @harding!

harding commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,186 @@
+      transport for the next packet to send, and finally calls `m_sock->Send()`."
+  a5link="https://bitcoincore.reviews/28165#l-83"
+
+  q6="How many bytes are sent over the wire for the `sendtxrcncl`message
```
Nit: can we mention what `sendtxrcncl` message does?  I'm parsing it as SendTransactionCancel, but I doubt that's correct.

murchandamus replied:
Yeah, it seems that it is “received from a peer demonstrating readiness to announce transactions via reconciliations.”

harding reviewed (commented):
A couple quick notes for @LarryRuane, with the overall text looking good.

LarryRuane commented on _posts/en/newsletters/2023-09-13-newsletter.md:
```diff
@@ -0,0 +1,186 @@
+  might even be required to support three or more different protocols
+  simultaneously, to the disadvantage of all.
+
+    Sander's summary had not received any comments as of this writing.
```
```suggestion
    Sander's summary has not received any comments as of this writing.
```
This comment is going to set the nittyness record, but the verb "had" slightly implies that it did receive comments _after_ the time of this writing.

LarryRuane commented:
Force pushed to resolve Dave's comments, thanks! (Even though I asked you a follow-up question, I decided to just go ahead and make the suggested changes.)

LarryRuane commented:
Force pushed for Murch's comment, good catch, thanks!

harding commented:
Pushed edits for all review comments (thanks @bitschmidty @murchandamus @LarryRuane!), added lede, checked for updates to the releases/RCs (no updates), and reviewed contributions (including updates) from @LarryRuane and @murchandamus (thanks!).  Also added topic entries.  Thanks everyone!

bitschmidty reviewed (commented):
ACK fixups

bitschmidty commented:
Squashed and merged 🚀
